### PR TITLE
Upgrade to wildfly CR2, re-add embedded, remove common-core dep

### PR DESCRIPTION
**Pull request by ctomc**
Also change the way we do testing with usage of provisioning plugin & feature-packs
should address https://issues.jboss.org/browse/WFLY-4572

### CODE REVIEW

**Comment by lfryc** [comment]
:+1: embedded container works great

File: common/pom.xml [hunk 1]
```diff
@@ -90,7 +91,7 @@
         <dependency>
             <groupId>org.jboss.sasl</groupId>
             <artifactId>jboss-sasl</artifactId>
-        </dependency>
+        </dependency>-->
```
Comment (bstansberry) [inline common/pom.xml]: If they aren't needed, let's delete.

File: container-managed-domain/pom.xml [hunk 2]
```diff
@@ -62,6 +59,13 @@
             <artifactId>javax.inject</artifactId>
         </dependency>
 
+        <!-- only here to satisfy two logging statements -->
+        <dependency>
+            <groupId>org.wildfly.core</groupId>
+            <artifactId>wildfly-server</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (bstansberry) [inline container-managed-domain/pom.xml]: What's needed to make this go away?

File: common/pom.xml (hunk 1 above)
Comment (ctomc) [inline common/pom.xml]: ah i missed that, let me push update

File: container-managed-domain/pom.xml (hunk 2 above)
Comment (ctomc) [inline container-managed-domain/pom.xml]: Use custom logging messages or copy the two localized ones from wildfly-server

**Comment by jamezp** [comment]
The provisioning stuff shouldn't be needed. We should be able to just remove the `org.wildfly.core:wildfly-core-feature-pack` dependencies which will actually resolve the issue from the JIRA.

**Comment by ctomc** [comment]
wdym @jamezp I did remove org.wildfly.core:wildfly-core-feature-pack dependancies.
provisioning is there just so we can run testsuite with valid WildFly distro

**Comment by bstansberry** [comment]
This should use core 1.0.0.CR2 once it's out on Friday. Then https://github.com/wildfly/wildfly-arquillian/blob/master/container-embedded/src/main/java/org/jboss/as/arquillian/container/embedded/EmbeddedDeployableContainer.java#L47 will need to be changed to not pass in a bundlePath param.

File: container-managed-domain/pom.xml (hunk 2 above)
Comment (ctomc) [inline container-managed-domain/pom.xml]: fixed

**Comment by ctomc** [comment]
@bstansberry I don't mind waiting till friday, but we should do release of this right after CR2 is out, so we can get it also to WildFly full.

File: container-managed-domain/pom.xml (hunk 2 above)
Comment (ctomc) [inline container-managed-domain/pom.xml]: fixed, by copying log messages from wildfly-server code

**Comment by jamezp** [comment]
@ctomc I don't think we need the provisioning either. I don't see any reason either the feature-pack or provisioning stuff is needed.

**Comment by lfryc** [comment]
Thanks guys!

**Pull request merged**